Kitodo-DataAccess/src/test/java/org/kitodo/dataaccess/format/xml/XMLWriterTest.java

```diff
@@ -0,0 +1,123 @@
+                .add(new MemoryNode(METS_DMD_SEC).put(METS_ID, "metadata").add(
+                    new MemoryNode(METS_MD_WRAP).put(METS_MDTYPE, "MODS").add(new MemoryNode(METS_XML_DATA).add(
+                        new MemoryNode(MODS_MODS).add(new MemoryNode(MODS_TITLE_INFO).add(
+                            new MemoryNode(MODS_TITLE).add(new MemoryLiteral("Hello World!", RDF.PLAIN_LITERAL))))))))
+                .add(new MemoryNode(METS_FILE_SEC).add(new MemoryNode(METS_FILE_GRP).put(METS_USE, "DEFAULT")
+                        .add(new MemoryNode(METS_FILE).put(METS_ID, "image").put(METS_MIMETYPE, "image/jpeg")
+                                .add(XLink
+                                        .createSimpleLink(MemoryStorage.INSTANCE, Optional.empty(),
+                                            "http://data.example.org/images/hello-world.jpg", Collections.emptySet(),
+                                            Collections.emptySet(), Optional.empty(), Optional.empty())
+                                        .put(RDF.TYPE, METS_F_LOCAT).put(METS_LOCTYPE, "URL")))))
+                .add(new MemoryNode(METS_STRUCT_MAP).put(METS_TYPE, "LOGICAL")
+                        .add(new MemoryNode(METS_DIV).put(METS_ID, "greeting").put(METS_DMDID, "metadata")
+                                .put(METS_LABEL, "Hello World!")))
+                .add(new MemoryNode(METS_STRUCT_MAP).put(METS_TYPE, "PHYSICAL")
+                        .add(new MemoryNode(METS_DIV).put(METS_TYPE, "physSequence")
+                                .add(new MemoryNode(METS_DIV).put(METS_ID, "page").put(METS_ORDER, "1")
+                                        .put(METS_ORDERLABEL, " - ").put(METS_TYPE, "page")
+                                        .add(new MemoryNode(METS_FPTR).put(METS_FILEID, "image")))))
+                .add(new MemoryNode(METS_STRUCT_LINK).add(XLink
+                        .createArcLink(MemoryStorage.INSTANCE, Optional.<String>of("greeting"), Optional.empty(),
+                            Optional.of("page"), Collections.emptySet(), Optional.empty(), Optional.empty())
+                        .put(RDF.TYPE, METS_SM_LINK)));
+        Namespaces namespaces = new Namespaces() {
+            private static final long serialVersionUID = 1L;
+            {
+                put("mets", "http://www.loc.gov/METS/");
+                put("mods", "http://www.loc.gov/mods/v3#");
+                put("xlink", "http://www.w3.org/1999/xlink#");
+            }
+        };
+        File expectedResult = new File("src/test/resources/xmlWriterTest-expectedResult.xml");
+        File testfile = File.createTempFile("xmlWriterTest-", ".xml");
+        try {
+            SerializationFormat.XML.write(data, namespaces, testfile);
+            assertThat(FileUtils.contentEquals(testfile, expectedResult), is(true));
+        } finally {
+            if (testfile.exists()) {
```

Comment:
Maybe you should move part of your code to `@Before` and `@After` methods to initialise and clean up your file space usage.

Reply:
Why? The code is specific to *this* test. `@Before` and `@After` methods will be executed before and after *every* test. If ever another test will be added (imagine something, `xmlWriter.isProperlyConfigured()`) the test file would be created and deleted, which does not make sense.